src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
+     * @for Fms
+     * @method _initializeArrivalRunway
+     * @private
      */
-    getBottomAltitude() {
-        const valueToExclude = INVALID_NUMBER;
-        const minAltitudeFromLegs = _without(
-            _map(this.legCollection, (leg) => leg.getProcedureBottomAltitude()),
-            valueToExclude
-        );
+    _initializeArrivalRunway() {
+        if (this.arrivalAirportModel === null) {
+            return;
+        }
 
-        return Math.min(...minAltitudeFromLegs);
-    }
+        const arrivalRunwayName = this._routeModel.getArrivalRunwayName();
 
-    /**
-     * Get the flight plan route string in dot notation
-     *
-     * @for Fms
-     * @method getFlightPlanRouteStringWithDots
-     * @return {string}
-     */
-    getFlightPlanRouteStringWithDots() {
-        const currentRouteSegments = _map(this.legCollection, (legModel) => legModel.routeString);
-        const previousAndCurrentRouteSegments = this._previousRouteSegments.concat(currentRouteSegments);
+        if (arrivalRunwayName === null) {
+            return this.setArrivalRunway(this.arrivalAirportModel.arrivalRunwayModel);
+        }
+
+        const arrivalRunwayModel = this.arrivalAirportModel.getRunway(arrivalRunwayName);
 
-        return previousAndCurrentRouteSegments.join('..').toUpperCase();
+        this.setArrivalRunway(arrivalRunwayModel);
     }
 
     /**
-     * Get the flight plan route string with legs separated by spaces
-     *
-     * This is primarily meant for use in the `StripViewModel`.
+     * Initialize #departureAirportModel
      *
      * @for Fms
-     * @method getFlightPlanRouteStringWithSpaces
-     * @return {string}
+     * @method _initializeDepartureAirport
+     * @param originIcao {string} ICAO identifier specified by spawn pattern
+     * @private
      */
-    getFlightPlanRouteStringWithSpaces() {
-        return this.getFlightPlanRouteStringWithDots().replace(REGEX.DOUBLE_OR_SINGLE_DOT, ' ');
+    _initializeDepartureAirport(originIcao) {
+        if (originIcao === '') {
+            return;
+        }
+
+        this.departureAirportModel = AirportController.airport_get(originIcao);
     }
 
     /**
-     * Get the next waypoint in the flight plan, if it exists
+     * Initialize #departureRunwayModel
      *
      * @for Fms
-     * @method getNextWaypointModel
-     * @return waypointModel {WaypointModel}
+     * @method _initializeDepartureRunway
+     * @private
      */
-    getNextWaypointModel() {
-        if (!this.hasNextWaypoint()) {
-            return null;
+    _initializeDepartureRunway() {
+        if (this.departureAirportModel === null) {
+            return;
         }
 
-        let waypointModel = this.currentLeg.nextWaypoint;
+        const departureRunwayName = this._routeModel.getDepartureRunwayName();
 
-        if (!this.currentLeg.hasNextWaypoint()) {
-            waypointModel = this.legCollection[1].currentWaypoint;
+        if (departureRunwayName === null) {
```

Comment:
`departureRunwayName === null` should be `!departureRunwayName`

_this is the last one I'll point out.  you get the idea 😉_